Aggregator/Aggregator.cs

```diff
@@ -0,0 +1,336 @@
+        protected override async Task RunAsync(CancellationToken cancellationToken)
+        {
+   
+            this.token = cancellationToken;
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                
+                await Task.Delay(TimeSpan.FromMilliseconds(SFUtilities.intervalMiliseconds), cancellationToken);
+                while(await GetMinQueueCountAsync() > 1)
+                {
+                    await ProduceSnapshotAsync();
+                    await TryPruneAsync(); 
+                }
+            }
+        }
+        public async Task PutDataRemote(string queueName,byte[] data)
+        {
+            await AddDataAsync(queueName, data);
+        }
+
+        public async Task<List<byte[]>> GetDataRemote(string queueName)
+        {
+            return await GetDataAsync(queueName);
+        }
+
+        
+        public async Task<List<Snapshot>> GetSnapshotsRemote(double milisecondsLow, double milisecondsHigh)
+        {
+            List<Snapshot> list = new List<Snapshot>();
+
+            var stateManager = this.StateManager;
+            var reliableQueue = await stateManager.GetOrAddAsync<IReliableQueue<byte[]>>(Snapshot.queueName);
+            try
+            {
+
+                using (var tx = stateManager.CreateTransaction())
+                {
+                    var iterator = (await reliableQueue.CreateEnumerableAsync(tx)).GetAsyncEnumerator();
+
+                    byte[] data = null;
+                    while (await iterator.MoveNextAsync(token))
+                    {
+                        data = iterator.Current;
+
+                        if (data != null) {
+                            Snapshot s = (Snapshot)ByteSerialization.ByteArrayToObject(data);
+                            if (s.miliseconds >= milisecondsLow && s.miliseconds <= milisecondsHigh) list.Add(s);
+                            else 
+                            {//help garbage collector to free memory
+                                s = null;
+                                data = null;
+                            }
+                        } 
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                var x = e;
+            }
+
+            return list;
+        }
+        /// <summary>
+        /// ClearAsync in not implemented for the RealiableQueue?!
+        /// Workaround would be to dequeue each element as implemented for pruning
+        /// </summary>
+        /// <returns></returns>
+        public async Task DeleteAllSnapshotsRemote()
+        {
+            using (var tx = this.StateManager.CreateTransaction())
+            {
+                try
+                {
+                    await (await this.StateManager.GetOrAddAsync<IReliableQueue<byte[]>>(Snapshot.queueName)).ClearAsync();
+                    await tx.CommitAsync();
+                }
+                catch(Exception e)
```

Comment:
Exception handling in general needs some work. This is a prototype, I understand that. Still, we should have some error handling in this codebase.

Reply:
This function isn't used, the ClearAsync method isn't implemented for the reliable queue. Workaround is to dequeue elements one by one as used in pruning. I left this here if the method gets implemented in the future so it's just a simple call to clear all snapshots.

Reply:
But yes exception handling does need some work, but it's never triggered for now.